scalameta/scalameta/shared/src/test/scala/scala/meta/tests/parsers/XmlSuite.scala

```diff
@@ -0,0 +1,298 @@
+      |        EOF [108..108) ----> class scala.meta.tokens.Token$EOF
+    """.stripMargin
+  )
+
+  test("deconstruct") {
+    val parsedTricky = term(trickyXml)
+    // format: off
+    val Term.Block(
+    Seq(
+    Defn.Val(Nil, Seq(Pat.Var.Term(Term.Name("x"))), None,
+    Term.Xml(Seq(Lit("<div href="), Lit(">Hello "), Lit("</div>")),
+    Seq(Term.ApplyInfix(Lit("/"), Term.Name("+"), Nil, Seq(Term.Name("url"))), Term.Name("name")))),
+    Defn.Val(Nil, Seq(Pat.Var.Term(Term.Name("noSemicolon"))), None,
+    Term.Xml(Seq(Lit("<h1>"), Lit("</h1>")),
+    Seq(Term.ApplyInfix(Term.Name("msg"), Term.Name("infix"), Nil, Seq(Term.Name("upper")))))),
+    Defn.Val(Nil, Seq(Pat.Var.Term(Term.Name("y"))), None, Lit(2)))) = parsedTricky
+    // format: on
+  }
+
+  checkOK("<foo/>")
+  checkOK("<foo></foo>")
+  checkOK("<foo/><bar/>")
+  checkOK("<foo:bar/>")
+  checkOK("<foo><bar/></foo>")
+  checkOK("<foo\n>\t <bar   />\t  </foo\n>")
+  checkOK("""<foo a="a" b="b"/>""")
+  checkOK("""<foo a:a="a" b:b="b"/>""")
+  checkOK("""<foo a="'"/>""")
+  checkOK("""<foo a='"'/>""")
+  checkOK("<foo>&name;</foo>")
+  checkOK("<foo>&lt;</foo>")
+  checkOK("<foo>Hello &name;!</foo>")
+  checkOK("""<foo a="&name;" />""")
+  checkOK("""<foo a="&lt;" />""")
+  checkOK("""<foo a="Hello &name;!"/>""")
+  checkOK("""<foo a="1 &lt; 2"/>""")
+  checkOK("<foo>&#1234;</foo>")
+  checkOK("<foo>&#x1234;</foo>")
+  checkOK("<foo>Hello&#x1234;Allan</foo>")
+  checkOK("""<foo a="&#1234;" />""")
+  checkOK("""<foo a="&#x1234;" />""")
+  checkOK("""<foo a="Hello&#x1234;Allan" />""")
+  checkOK("<xml:group><foo/><bar/></xml:group>")
+  checkOK("<xml:group></xml:group>")
+  checkOK("<!---->")
+  checkOK("<!----->")
+  checkOK("<!--foo-->")
+  checkOK("<!--a-b-->")
+  checkOK("<![CDATA[foo]]>")
+  checkOK("<![CDATA[]]>")
+  checkOK("<![CDATA[>]]>")
+  checkOK("<![CDATA[]>]]>")
+  checkOK("<![CDATA[]]]]>")
+  checkOK("<?foo bar?>")
+  checkOK("<?foo?>")
+  checkOK("<?foo     ?>")
+  checkOK("<xml:unparsed>foo</xml:unparsed>")
+  checkOK("<xml:unparsed>{</xml:unparsed>")
+  checkOK("<xml:unparsed><</xml:unparsed>")
+  checkOK("<![CDATA[hello, world]]>")
+  checkOK("<![CDATA[hello, world]]><![CDATA[hello, world]]>")
+  checkOK("<foo>x<![CDATA[hello, world]]></foo>")
+  checkOK("<foo><![CDATA[hello, world]]></foo>")
+  checkOK("<foo><![CDATA[hello, world]]><![CDATA[hello, world]]></foo>")
+  checkOK("<a><b/>start<![CDATA[hi & bye]]><c/>world<d/>stuff<![CDATA[red & black]]></a>")
+  checkOK("<foo>{foo}</foo>")
+  checkOK("<foo>{foo }</foo>")
+  checkOK("<foo>{ foo}</foo>")
+  checkOK("<foo>{ foo }</foo>")
+  checkOK("<foo>{2}</foo>")
+  checkOK("""<foo>{"bar"}</foo>""")
+  checkOK("<foo>{1}</foo>")
+  checkOK("<foo>{<bar/>}</foo>")
+  checkOK("<foo>{<bar/><bat/>}</foo>")
+  checkOK("""<foo a={"foo"}/>""")
+  checkOK("<foo a={<bar/>}/>")
+  checkOK("<foo a={<bar/><bat/>}/>")
+  checkOK("<a>{ List(1, 2) }</a>")
+  checkOK("<a>{1}{2}</a>")
```

Comment:
Yes, please. This is what string interpolation tokenizer is doing.

Would you prefer to do this differently?